modules/core/src/main/java/org/apache/fluo/core/client/FluoClientImpl.java

```diff
@@ -49,19 +52,17 @@ public static final AutoCloseable setupReporters(Environment env, String id,
         + reporterCounter.getAndIncrement());
   }
 
-  public FluoClientImpl(FluoConfiguration config) {
-    this.config = config;
-    if (!config.hasRequiredClientProps()) {
-      String msg = "Client configuration is missing required properties";
-      log.error(msg);
-      throw new IllegalArgumentException(msg);
-    }
+  public FluoClientImpl(FluoConfiguration connConfig) {
+    Objects.requireNonNull(connConfig);
+    Preconditions.checkArgument(connConfig.hasRequiredConnectionProps(),
+        "missing required connection properties");
+    config = FluoAdminImpl.mergeZookeeperConfig(connConfig);
```

Comment:
I think this call creates a curator and closes it.  I suspect later in this method a curator is created, could possibly refactor code to avoid creating multiple connections to ZK.